contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,751 @@
+        delete dependencyType.scriptBytecodeAddresses[
+            dependencyType.scriptCount - 1
+        ];
+        unchecked {
+            dependencyType.scriptCount = dependencyType.scriptCount - 1;
+        }
+
+        emit DependencyTypeScriptUpdated(_dependencyTypeId);
+    }
+    
+    /**
+     * @notice Updates preferred CDN for dependency type `_dependencyTypeId`.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _preferredCDN URL for preferred CDN.
+     */
+    function updateDependencyTypePreferredCDN(
+        bytes32 _dependencyTypeId,
+        string memory _preferredCDN
+    )
+        external
+        onlyAdminACL(this.updateDependencyTypePreferredCDN.selector)
+        onlyNonEmptyString(_preferredCDN)
+    {
+        dependencyTypeInfo[_dependencyTypeId].preferredCDN = _preferredCDN;
+
+        emit DependencyTypePreferredCDNUpdated(_dependencyTypeId, _preferredCDN);
+    }
+
+    /**
+     * @notice Updates preferred repository for dependency type `_dependencyTypeId`.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _preferredRepository URL for preferred repository.
+     */
+    function updateDependencyTypePreferredRepository(
+        bytes32 _dependencyTypeId,
+        string memory _preferredRepository
+    )
+        external
+        onlyAdminACL(this.updateDependencyTypePreferredRepository.selector)
+        onlyNonEmptyString(_preferredRepository)
+    {
+        dependencyTypeInfo[_dependencyTypeId].preferredRepository = _preferredRepository;
+
+        emit DependencyTypePreferredRepositoryUpdated(
+            _dependencyTypeId,
+            _preferredRepository
+        );
+    }
+
+    /**
+     * @notice Updates project website for dependency type `_dependencyTypeId`.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _projectWebsite URL for project website.
+     */
+    function updateDependencyTypeProjectWebsite(
+        bytes32 _dependencyTypeId,
+        string memory _projectWebsite
+    ) external onlyAdminACL(this.updateDependencyTypeProjectWebsite.selector) {
+        dependencyTypeInfo[_dependencyTypeId].projectWebsite = _projectWebsite;
+
+        emit DependencyTypeProjectWebsiteUpdated(
+            _dependencyTypeId,
+            _projectWebsite
+        );
+    }
+
+    /**
+     * @notice Adds a new CDN url to dependencyType `_dependencyTypeId`.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _additionalCDN CDN URL to be added. Required to be a non-empty string,
+     * but no further validation is performed.
+     */
+    function addDependencyTypeAdditionalCDN(
+        bytes32 _dependencyTypeId,
+        string memory _additionalCDN
+    )
+        external
+        onlyAdminACL(this.addDependencyTypeAdditionalCDN.selector)
+        onlyNonEmptyString(_additionalCDN)
+    {
+        uint24 additionalCDNCount = dependencyTypeInfo[_dependencyTypeId]
```

Comment:
nit: might benefit from loading `dependencyTypeInfo[_dependencyTypeId]` to a storage pointer here